src/components/MessageBar/MessageBar.jsx

```diff
@@ -10,35 +19,138 @@ import {
   SendIcon,
 } from './MessageBarStyled';
 
-const MessageBar = () => {
-  const [value, setValue] = useState('');
+const MessageBar = ({
+  countryData,
+  setCountryData,
+  setSubscriptionCountryRooms,
+}) => {
+  const [message, setMessage] = useState('');
+  const [isShowJoinBtn, setIsShowJoinBtn] = useState(true);
+  const userId = useSelector(getUser)?.id;
   const textAreaRef = useRef(null);
+  const stompClient = useStompClient();
+  const { sendMessage } = useWebSocket();
 
-  const isInputNotEmpty = Boolean(value?.trim().length);
+  const isInputNotEmpty = Boolean(message?.trim().length);
 
-  const handleChange = e => {
-    setValue(e.target.value);
+  const handleChange = ({ target: { value } }) => setMessage(value);
+
+  useEffect(() => {
```

Comment:
You don't need this useEffect and const [isShowJoinBtn, setIsShowJoinBtn] = useState(true) as well. Just use countryData?.isSubscribed